solutions/devsprint-1/DeliveryAppChallenge/ViewModels/RestaurantsViewModel.swift

```diff
@@ -0,0 +1,67 @@
+//
+//  RestaurantsViewModel.swift
+//  DeliveryAppChallenge
+//
+//  Created by Dairan on 09/11/21.
+//
+
+import Foundation
+
+// MARK: - RestaurantsViewModel
+
+class RestaurantsViewModel {
+    // MARK: Lifecycle
+
+    init(repository: DeliveryApi = DeliveryApi()) {
+        self.repository = repository
+        fetchData()
+    }
+
+    // MARK: Internal
+
+    var bindViewModelUpdated: (() -> Void)?
+
+    private(set) var restaurants: [Restaurant] = [] {
+        didSet {
+            bindViewModelUpdated?()
+        }
+    }
+
+    var numberOfRows: Int {
+        return restaurants.count
+    }
+
+    func itemForCell(at index: IndexPath) -> RestaurantCell {
+        let restaurants = restaurants[index.row]
+        return RestaurantCell(restaurant: restaurants)
+    }
+
+    // MARK: Private
+
+    private let repository: DeliveryApi
+
+    private func fetchData() {
+        repository.fetchRestaurants { result in
+            switch result {
+                case let .success(restaurants):
+                    self.restaurants = restaurants
+                case let .failure(error):
+                    print(error)
+            }
+        }
+    }
+}
+
+// MARK: - RestaurantCell
+
+struct RestaurantCell {
```

Comment:
Creio que essa camada da Cell poderia ser feito em outra área como no exemplo do DataSource que havia mencionado.

Reply:
Eu não vou conseguir fazer isso agora, conforme mencionei no Discord. Preciso entender melhor o funcionamento. Mas se for para copiar e colar eu faço. ;)